Reduce memory usage at rest

mstoeckl opened the pull request:
This PR partially addresses #14. See commit messages for details.

There is a new dependency on gio-unix-2.0, to provide the 20 necessary lines of code from [`g_unix_input_stream_new`](https://github.com/frida/glib/blob/master/gio/gunixinputstream.c). This can be removed (at +38/-12 diff cost) if the code switches to using GdkPixbufLoader directly.

emersion commented:
Overall this looks pretty good! I do wonder though whether loading the images from a FD is really beneficial. The inode matching allows us to de-duplicate symlinks, but comparing paths doesn't sound too bad and should account for 99% of the users. Maybe we should consider dropping the first commit?

emersion commented on main.c:
```diff
@@ -44,12 +44,23 @@ struct swaybg_state {
 struct swaybg_output_config {
 	char *output;
-	cairo_surface_t *image;
+	struct swaybg_image *image;
```
Instead of ref'counting images, would it be simpler to just store a `char *path` at config-parse-time, and then load all images in one go once we've decided the output <-> image mapping?

mstoeckl replied:
Yes; changing this only affects the error messages in edge cases like 'swaybg -i file_not_found.png -i file_exists.png'.

mstoeckl commented:
> I do wonder though whether loading the images from a FD is really beneficial

The main benefits were both edge cases:
* Being able to deduplicate soft and hard links, as well as unusual paths, like `a.png` vs `./a.png`
* Only needing to call `open()` once per image; instead of e.g. once per monitor hotplug. This ensured that the background could still be drawn if the image file was deleted or renamed before a new output was added. 

> Maybe we should consider dropping the first commit?

Dropped, and adjusted following commits. The FD loading can always be added in later, if anyone wants it.

emersion reviewed (approved):
LGTM, thanks!
